### ISSUE TITLE
SPManagedMetaDataServiceApp Resource shows error when running DSC

### ISSUE THREAD

**Issue by Bjorn80** [PowerShell/SharePointDsc#453]
**Details of the scenario you try and problem that is occurring:**
The SPManagedMetaDataServiceApp creates the Managed Metadata Service Application successfully, but when rerunning start-dscconfiguraion -useexisting -verbose -wait -force and update-dscconfiguration -verbose -force or timed dsc run, I get the following errors:

```
VERBOSE: [vm06]: LCM:  [ Start  Resource ]  [[SPServiceInstance]ManagedMetadataServiceInstance]
VERBOSE: [vm06]: LCM:  [ Start  Test     ]  [[SPServiceInstance]ManagedMetadataServiceInstance]
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Testing service instance 'Managed Metadata Web Service'
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Getting service instance 'Managed Metadata Web Service'
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Executing as the local run as user contoso\SPInstall
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Leaving BeginProcessing Method of Get-SPServiceInstance.
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Leaving ProcessRecord Method of Get-SPServiceInstance.
VERBOSE: [vm06]:                            [[SPServiceInstance]ManagedMetadataServiceInstance] Leaving EndProcessing Method of Get-SPServiceInstance.
VERBOSE: [vm06]: LCM:  [ End    Test     ]  [[SPServiceInstance]ManagedMetadataServiceInstance]  in 2.1400 seconds.
VERBOSE: [vm06]: LCM:  [ Skip   Set      ]  [[SPServiceInstance]ManagedMetadataServiceInstance]
VERBOSE: [vm06]: LCM:  [ End    Resource ]  [[SPServiceInstance]ManagedMetadataServiceInstance]
VERBOSE: [vm06]: LCM:  [ Start  Resource ]  [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp]
VERBOSE: [vm06]: LCM:  [ Start  Test     ]  [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp]
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Testing managed metadata service application Managed Metadata Service Application
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Getting managed metadata service application Managed Metadata Service Application
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Executing as the local run as user contoso\SPInstall
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving BeginProcessing Method of Get-SPServiceApplication.
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving ProcessRecord Method of Get-SPServiceApplication.
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving EndProcessing Method of Get-SPServiceApplication.
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving BeginProcessing Method of Get-SPServiceApplicationProxy.
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving ProcessRecord Method of Get-SPServiceApplicationProxy.
VERBOSE: [vm06]:                            [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp] Leaving EndProcessing Method of Get-SPServiceApplicationProxy.
VERBOSE: [vm06]: LCM:  [ End    Test     ]  [[SPManagedMetaDataServiceApp]ManagedMetadataServiceApp]  in 1.8130 seconds.
PowerShell DSC resource MSFT_SPManagedMetaDataServiceApp  failed to execute Test-TargetResource functionality with error message: You cannot call a method on a null-valued expression. 
    + CategoryInfo          : InvalidOperation: (:) [], CimException
    + FullyQualifiedErrorId : ProviderOperationExecutionFailure
    + PSComputerName        : localhost  
```

and at the bottom:

```
The SendConfigurationApply function did not succeed.
    + CategoryInfo          : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 1
    + PSComputerName        : localhost 
```

If i remove the resource-block and run, no errors occur.

My SharePoint configuration setup is 1 app server and 2 WFE . The two WFE run the same configuration where they both first start the service instance for the service, then the SPManagedMetaDataServiceApp  block.

I have the same result on a mirror environment (same setup).
Any ideas what may be causing this?
The Service Managed Service application seems fine.

**The DSC configuration that is using the resource:**
```
        SPManagedMetaDataServiceApp ManagedMetadataServiceApp
        {  
            Name                 = "Managed Metadata Service Application"
            PsDscRunAsCredential = $SPSetupCred
            ApplicationPool      = $ConfigurationData.NonNodeData.SharePoint.ApplicationPool
            DatabaseServer       = $ConfigurationData.NonNodeData.SharePoint.SqlAlias
            DatabaseName         = $ConfigurationData.NonNodeData.SharePoint.ManagedMetadataService.DatabaseName
            DependsOn            = "[SPServiceInstance]ManagedMetadataServiceInstance"
        }  
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Server 2012R2 and PowerShell 5

**Version of the DSC module you're using:**
SharePointDSC 1.4.0.0

**Comment by luigilink** [comment]
This issue is already fixed in PR #446
Issue Ref : https://github.com/PowerShell/SharePointDsc/issues/445

**Comment by BrianFarnhill** [comment]
Thanks @luigilink - as stated this is a duplicate so closing this off. The fix will be released in 1.5 and is already in the dev branch.
